components/brave_wallet/browser/tx_service.cc

```diff
@@ -51,18 +68,37 @@ size_t CalculatePendingTxCount(
 TxService::TxService(JsonRpcService* json_rpc_service,
                      BitcoinWalletService* bitcoin_wallet_service,
                      KeyringService* keyring_service,
-                     PrefService* prefs)
+                     PrefService* prefs,
+                     const base::FilePath& context_path)
     : prefs_(prefs), json_rpc_service_(json_rpc_service), weak_factory_(this) {
-  tx_manager_map_[mojom::CoinType::ETH] = std::make_unique<EthTxManager>(
-      this, json_rpc_service, keyring_service, prefs);
-  tx_manager_map_[mojom::CoinType::SOL] = std::make_unique<SolanaTxManager>(
-      this, json_rpc_service, keyring_service, prefs);
-  tx_manager_map_[mojom::CoinType::FIL] = std::make_unique<FilTxManager>(
-      this, json_rpc_service, keyring_service, prefs);
+  store_factory_ = base::MakeRefCounted<value_store::ValueStoreFactoryImpl>(
+      context_path.AppendASCII(kWalletBaseDirectory));
+  store_ = std::make_unique<value_store::ValueStoreFrontend>(
+      store_factory_, base::FilePath(kWalletStorageName),
+      kValueStoreDatabaseUMAClientName,
+#if BUILDFLAG(IS_IOS)
+      web::GetUIThreadTaskRunner({}),
+#else
+      content::GetUIThreadTaskRunner({}),
+#endif
```

Comment:
Can we move these to TxServiceFactory? 
So we can nicely get rid of IS_IOS check and more importantly don't need to make brave_wallet a content dependent component which may trigger issues with ios support in the future

Reply:
fixed in 6505b8dd6397978a8667a4ca135829a53a256b44